salt/metalk8s/orchestrate/upgrade/etcd.sls

```diff
@@ -0,0 +1,52 @@
+{%- set dest_version = pillar.orchestrate.dest_version %}
+{%- set etcd_nodes = salt.metalk8s.minions_by_role('etcd') %}
+{%- set control_plane_ip = salt.saltutil.runner(
+    'mine.get',
+    tgt=pillar.bootstrap_id,
+    fun='control_plane_ip'
+)[pillar.bootstrap_id] %}
+
+{%- set pillar_data = {
+    'repo': {
+        'host': control_plane_ip
+    },
+    'registry_ip': control_plane_ip
+} %}
+
+Check etcd cluster health:
+  module.run:
+    - metalk8s.check_etcd_health:
+      - hostname: {{ pillar.bootstrap_id }}
+
+{%- for node in etcd_nodes | sort %}
+
+Sync {{ node }} minion:
+  salt.function:
+    - name: saltutil.sync_all
+    - tgt: {{ node }}
+    - saltenv: metalk8s-{{ dest_version }}
```

Comment:
I think this should also require the node to be in 'ok' state (`require: ['module: Check etcd cluster health']`). There's no need to sync modules if we're not going to perform the upgrade in the next step anyway.

Reply:
For me it's not a requirements you don't need an healthy etcd cluster to sync modules and if it's just sync at a failing run at the next run he will basically do nothing

Reply:
I think that would still need to be done, since we use `require`-chain to define the order of execution, not just to define "real" requirements.